### PR TITLE
Add and remove

### PR DESCRIPTION
**Pull request by mukhammadrizooff**
# In This Branch I Add :
- **Removed all hardcoded items from the tasks array.**
- **Created a new JavaScript file for the new functionality.**
- **I implement a function for adding a new task.**
- **I implement a function for deleting a task.**
- **I implement a function for editing task descriptions.**
- **By default new tasks have the property completed set to false and the property index set to the value of the new array length.**
- **Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.**
- **All changes to the To Do List saved in local storage.**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 Additional description about the project and its features.
 
-![2022-01-19_21-05-17](https://user-images.githubusercontent.com/63915024/150205559-927409bd-a533-44b4-ad77-02bbacbac2bd.png)
+![2022-01-20_19-37-00](https://user-images.githubusercontent.com/63915024/150359553-4e2fcd2a-4533-44c4-8583-51ace5b2fae6.png)
 
 ## Built With
```
Comment (brytebee) [inline README.md:10]: - [x] **Optional:** Live demo link opens to the README page. Fix this bug.

File: src/modules/storage.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export default class StorageManager {
+  static save = (tasks) => {
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+  };
+}
```
Comment (brytebee) [inline src/modules/storage.js]: - [x] **Required:** There should also be a getter function for getting data from the local storage before feeding it to your `controlDisplay` to `displayTask`. This is the reason why every time the page reloads the previous data although there cannot be rendered to the page.

File: src/index.js [hunk 3]
```diff
@@ -1,36 +1,69 @@
...
+    selector('.line input').focus();
+  }
+};
+
+const removeList = (index) => {
+  manager.removeList(index);
+  ControlDisplay.reset(selector('.tasks'));
+  manager
+    .getTasks()
+    .forEach((task) => ControlDisplay.displayTask(selector('.tasks'), task));
```
Comment (brytebee) [inline src/index.js]: - [x] **Hint:** Before page render you should check if the local storage contains any data that should be rendered, else render new data supplied by the user. I hope this makes sense 😄😄

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @mukhammadrizooff👋,

Good job so far coming so far with the progress you've made! 💪💪

## Highlights
You have done amazing in the following:
- pretty UI design✔️
- Nice implementation of add and remove functionalities✔️
- awesome implementation of keyboard event✔️
- nicely organized code into modules✔️
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (mukhammadrizooff) [reply README.md:10]: ### @brytebee
### You tried the Live demo link by clicking view deployment in the PR section. I would ask you to go to this [Page](https://github.com/mukhammadrizooff/To-Do-List/tree/add-and-remove) and try it out from the link in my `README.md` file. It is fully functional.

File: src/modules/storage.js (hunk 2 above)
Comment (mukhammadrizooff) [reply src/modules/storage.js]: @brytebee Thank you I did what you said please take a re-review.

File: src/index.js (hunk 3 above)
Comment (mukhammadrizooff) [reply src/index.js]: @brytebee This is also done thank you. Can you take a re-review?

File: src/modules/display.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+export default class ControlDisplay {
```
Comment (petudeveloper) [inline src/modules/display.js:1]: Your application is working fine except for not rendering the data in your local storage after refreshing the page. Please make the necessary changes to correct this error.

Check the image below for a better understanding:
1. Image before refreshing the page, the data is saved in your local storage and rendered.
Screenshot from 2022-01-20 12-29-03

2. Image after refreshing your page, the data is still saved in your local storage but is not rendered.
Screenshot from 2022-01-20 12-30-02

**Review by petudeveloper — CHANGES_REQUESTED** [review]
Hi @mukhammadrizooff, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

very close

 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (brytebee) [reply README.md:10]: Okay.

**Review by andersonlebon — APPROVED** [review]
Hi @mukhammadrizooff 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is now professional :heavy_check_mark: 

## Optional suggestions
- [x] Kindly Check the comments under the review._
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by david-lafontant** [comment]
> * **Removed all hardcoded items from the tasks array.**
> * **Created a new JavaScript file for the new functionality.**
> * **I implement a function for adding a new task.**
> * **I implement a function for deleting a task.**
> * **I implement a function for editing task descriptions.**
> * **By default new tasks have the property completed set to false and the property index set to the value of the new array length.**
> * **Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.**
> * **All changes to the To Do List saved in local storage.**

rrrrrrrrrrrrrrrrrrrrrrr

**Pull request merged**